flask/cli.py

```diff
@@ -544,7 +544,10 @@ def main(self, *args, **kwargs):
         # script that is loaded here also attempts to start a server.
         os.environ['FLASK_RUN_FROM_CLI'] = 'true'
 
-        if self.load_dotenv:
+        val = os.environ.get('FLASK_DONT_LOAD_ENV')
+        load_dotenv =  not val or val in ('0', 'false', 'no')
```

Comment:
`load_dotenv` is the name of the function that will be called later, it's been overwritten as a boolean.